src/_posts/2019-01-07-mars-rover-kata-refactoring-to-patterns.md

```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+asset-type: post
+name: mars-rover-kata-refactoring-to-patterns
+title: Mars Rover Kata - Refactoring to Patterns
+date: 2019-01-08 07:20:00 +00:00
+author: Simion Iulian Belea
+description: An example of what is learned in the first few weeks of the apprenticeship with the Mars Rover kata.
+image: 
+    src: /assets/custom/img/blog/mars-rover.jpg
+tags: 
+- refactoring
+- patterns
+- apprenticeship
+- java
+- kata
+
+---
+
+
+
+# Design patterns and the apprenticeship
+
+This article is an account of how I learned applying patterns. It condenses concepts from the first two weeks of the apprenticeship program.
+
+A kata is a very good exercise where you&#39;re given a specific set of rules and then one can use that exercise to learn different concepts, as well as different programming languages. One has an isolated domain problem where he can seek a different solution everytime. This can lead to a better understanding of how to solve a specific set of problems.
```

Comment:
How about "One has an isolated problem domain where they can produce a different solution each time"?

Reply:
A kata is a very good exercise where you&#39;re given a specific set of rules and then one can use that exercise to learn different concepts, as well as different programming languages. One has an isolated domain problem where he can seek a different solution everytime. This can lead to a better understanding of how to solve a specific set of problems.